src/main/java/org/jitsi/jicofo/recording/jibri/JibriRecorder.java

```diff
@@ -619,13 +639,44 @@ synchronized public void onJibriStatusChanged(String jibriJid, boolean idle)
     }
 
     @Override
-    synchronized public void onJibriOffline(String jibriJid)
+    synchronized public void onJibriOffline(final String jibriJid)
     {
         if (jibriJid.equals(recorderComponentJid))
         {
             logger.warn("Our recorder went offline: " + recorderComponentJid);
 
-            recordingStopped(null);
+            // Check if we have Jibri available
+            final String newJibriJid = jibriDetector.selectJibri();
+            if (newJibriJid == null)
+            {
+                recordingStopped(null);
+                return;
+            }
+
+            XMPPError err = null;
+            for (int i = 0; i < NUM_RETRIES; i++)
+            {
+                try
+                {
+                    err = startJibri(newJibriJid);
+                }
+                catch (final SmackException.NoResponseException e)
+                {
+                    err = new XMPPError(XMPPError.Condition.request_timeout);
+                }
+
+                if (err == null) {
+                    break;
+                }
+
+                // Mask the original error in the IQ response.
```

Comment:
If we are masking the original error, can we at least log it for debugging purposes ?